orchestrator/cli/domain_gen_helpers/resource_type_helpers.py

```diff
@@ -0,0 +1,220 @@
+                prefilled_val = inputs.get(db_props[0], {}).get("update")
+                should_update = (
+                    prefilled_val
+                    if prefilled_val
+                    else get_user_input(
+                        f"Change resource type {db_props} to {model_props} (y/N): ",
+                        "n",
+                    )
+                )
+                if should_update == "y":
+                    updates[db_props[0]] = model_props[0]
+
+    for name, diff in block_diffs.items():
+        db_props = list(diff.get("missing_resource_types_in_model", set()))
+        model_props = list(diff.get("missing_resource_types_in_db", set()))
+
+        for key, value in updates.items():
+            if key in db_props and value in model_props:
+                db_props.remove(key)
+                model_props.remove(value)
+
+        block_diffs[name]["missing_resource_types_in_model"] = set(db_props)
+        block_diffs[name]["missing_resource_types_in_db"] = set(model_props)
+    return updates
+
+
+def generate_create_resource_types_sql(
+    resource_types: List[str], inputs: Dict[str, Dict[str, str]], revert: bool = False
+) -> List[str]:
+    def create_resource_type(resource_type: str) -> str:
+        if revert:
+            description = (
+                ResourceTypeTable.query.where(ResourceTypeTable.resource_type == resource_type)
+                .with_entities(ResourceTypeTable.description)
+                .one()
+            )[0]
+        else:
+            print(f"--- RESOURCE TYPE ['{resource_type}'] ---")  # noqa: T001, T201
+            description = inputs.get(resource_type, {}).get("description") or get_user_input(
+                "Resource type description: "
+            )
+
+        return sql_compile(
+            Insert(ResourceTypeTable).values({"resource_type": resource_type, "description": description})
+        )
+
+    return [create_resource_type(resource_type) for resource_type in resource_types]
+
+
+def generate_update_resource_types_sql(resource_types: Dict[str, str]) -> List[str]:
+    def update_resource_type(old_rt_name: str, new_rt_name: str) -> str:
+        return sql_compile(
+            Update(ResourceTypeTable)
+            .where(ResourceTypeTable.resource_type == old_rt_name)
+            .values(resource_type=new_rt_name)
+        )
+
+    return [update_resource_type(*item) for item in resource_types.items()]
+
+
+def generate_delete_resource_types_sql(resource_types: List[str]) -> List[str]:
+    if not resource_types:
+        return []
+    return [
+        sql_compile(
+            Delete(SubscriptionInstanceValueTable).where(
+                SubscriptionInstanceValueTable.resource_type_id.in_(get_resource_types(resource_types))
+            )
+        ),
+        sql_compile(Delete(ResourceTypeTable).where(ResourceTypeTable.resource_type.in_(resource_types))),
+    ]
+
+
+def generate_create_resource_type_relations_sql(
```

Comment:
split to function to create resource type relations and resource type instance relations.